modules/nginx/manifests/restart.pp

```diff
@@ -1,7 +1,7 @@
 # FIXME: This class needs better documentation as per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc
 class nginx::restart {
 
-  exec { '/etc/init.d/nginx configtest && /etc/init.d/nginx restart':
+  exec { '/etc/init.d/nginx configtest && /etc/init.d/nginx restart || {nginx -s quit; /etc/init.d/nginx start}':
```

Comment:
This will not work as expected/ run the nginx stop and start on failure of the configtest - maybe we can move the configtest into an onlyif?
```    
exec { '/etc/init.d/nginx restart || {nginx -s quit; /etc/init.d/nginx start}':
    onlyif     => '/etc/init.d/nginx configtest'
```
Or add more brackets ;)

Reply:
how I read it is:
1. run the configtest,
2. if successful, run the restart command
3. if restart command fails, run quit command and the nginx start.

Reply:
If I use the `onlyif`, this will revert what Reuben did here: https://github.com/alphagov/govuk-puppet/pull/8541/files

Reply:
I added more brackets and testes all scenarios on my bash

Reply:
Sry to be a pain, but this is still not working in bash for me - let's sit down together for a bit tomorrow to get this out of the door

Reply:
Works in situ - I only checked on my Mac's bash... cheers!